README.md

```diff
@@ -49,11 +49,10 @@ app$layout_set(
   htmlDiv(id = "outputID")
 )
 
-app$callback(
-  function(x = input("inputID"), y = input("inputID", "type")) {
+app$callback(output=list("outputID"), inputs=list(list("inputID", "type")), user_function=
```

Comment:
Also, should this be `output=list("outputID", "children")`?

Reply:
Also, for the pedagogical example, I think we should do `inputs=list(list("inputID", "value"))`, since that's more useful than `type` (`type` doesn't change during runtime, but `value` changes when the user types into the input)

Reply:
> Also, should this be `output=list("outputID", "children")`?

While the app seems to run without the modification, I think it makes sense to change this.

Reply:
I actually kind of like having to use a special function/constructor for the references here. So rather than `output=list("outputID", "children")` you'd just have `output("outputID", "children")`. Seems to me this has a few benefits:
- it'll immediately tell you if you forgot the `"children"` - and presumably the IDE can tell you as soon as you type `output(` that you need an id and a property. That said, I notice in the original code `"children"` is the default property (and `"value"` for input/state). I don't think that's a good idea, we should force users to be explicit about the property being referenced. (also there's still an `event` in that file)
https://github.com/plotly/dashR/blob/5f89e8034c793e2ae37266b58251887542280667/R/dependencies.R#L16
- More of the syntax conveys meaning, no extra `list`. You'd still have `list` for the inputs, but only the outer one: `list(input("inputID", "value"))`
- This would differ from the python side, but what if we combined inputs and state into a single list like `list(input("inputID", "value), state("input2ID", "value"))? We would need to check and throw an error if there isn't at least one `input` in the list, and we could discuss whether all the inputs need to be before all the states or if we can accept arbitrary order, but the advantage is then all arguments are required - you just have `output`, `inputs_and_state`, and `func` - so the user doesn't need to name any of them.

Reply:
Thanks for these suggestions; I like this approach as well. The `state` parameter should still be optional, correct? If true, I think we'd set no default for the `input` parameter, but use `NULL` as the default for `state`.

Reply:
I thought what we settled on at the meeting was to combine input and state into one parameter (that I was calling `inputs_and_state` but could just be `params` or something). Then we set no defaults for any of the parameters, as they're all required.

Reply:
Yes, as we discussed, this is a great suggestion. Will use class attributes to differentiate between inputs and state, and set them all in one shot.

Reply:
We still want to change this, right?
```R
app$callback(output("outputID", "children"), list(input("inputID", "value")), function(x) {
```

Reply:
I can simplify the example so that the formals don't include the argument names.